riscv-sbi.adoc

```diff
@@ -1674,6 +1675,259 @@ The possible error codes returned in `sbiret.error` are shown in the
+                                           unsigned long cert_request_size
+                                           unsigned long report_data_addr,
+                                           unsigned long evidence_format,
+                                           unsigned long cert_addr,
+                                           unsigned long max_cert_size);
+----
+
+Retrieve a TCB attestation evidence from the underlying SBI implementation.
+Both the constitution and definition of the TCB and its measurement are
+implementation specific.
+
+The caller provides a https://datatracker.ietf.org/doc/html/rfc5280[X.509]
+formatted Certificate Signing Request (CSR) at the `cert_request_addr` physical
+address, which length is `cert_request_size`.
+
+Additionally, the caller provides a 64 bytes data blob at the
+`report_data_addr` physical address. This blob will be included in the
+generated certificate and can be made of any arbitrary piece of data.
+As an example, it could be a cryptographic nonce that must be bound to the
+generated evidence in order to protect the remote attestation process from
+replay attacks.
+
+The SBI implemenation replies to the signing request by generating a
+https://datatracker.ietf.org/doc/html/rfc5280[X.509] formatted certificate at
+the `cert_addr` physical address. The generated certificate must be signed with
+a platform attestation keys and must include a hash of the data blob pointed by
+`report_data_addr`. Its actual length is in `sbiret.value` and must not exceed
+`max_cert_size` bytes
+
+Moreover the generated certificate must contain:
+
+* A X.509 extension formatted according to `evidence_format`, containing the
+attestation evidence that is associated with the CSR `Subject` key, i.e. the
+generated certificate `Subject` and `SubjectPublicKey` identify the entity
+associated with this extension's defined evidence.
+`evidence_format` must be one of the supported evidence formats, as reported by
+the `EVIDENCE_FORMATS` field from the <<get-capabilities-function>> returned
+structure (`ATST_CAPS`). See
+<<table_attestation_and_measurement_capabilities_format>> for more details.
+* A https://oidref.com/2.5.29.49[UserNotice] X.509 extension containing the
+the data blob from `report_data_addr`.
+* A https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.1[AuthorityKeyIdentifier]
+X.509 extension.
+
+The possible error codes returned in `sbiret.error` are shown in the
+<<table_get_evidence_errors>> below.
+
+[#table_get_evidence_errors]
+.Attestation Get Evidence Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code          | Description
+| SBI_SUCCESS         | The evidence is generated and the certificate size is
+                        `sbiret.value`.
+| SBI_ERR_FAILED      | The evidence could not be generated.
+| SBI_INVALID_ADDRESS | The CSR or the certificate adress is invalid.
+| SBI_INVALID_PARAM   | The CSR at `csr_addr` is invalid.
+|===
```

Comment:
| SBI_ERR_BUSY   | The request failed due to resource busy error and may be retried.